### ISSUE TITLE
Valid NetCDF files fail to upload with capital file extension

### ISSUE THREAD

**Issue by daltonkell** [ioos/compliance-checker-web#113]
Attempting to upload `Stoch_Images.NC` results in the following error:

![image](https://user-images.githubusercontent.com/32339201/88939386-77f74100-d254-11ea-8007-a91287509a1b.png)

The file is, in fact, an HDF5 file:

```
$ file Stoch_Images.NC 
Stoch_Images.NC: Hierarchical Data Format (version 5) data
```

The Compliance Checker CLI has no trouble checking the file.

This error is likely due to checking files purely based on extensions, where `.NC` (all-caps) is not allowed.

**Comment by Bobfrat** [comment]
you can test that theory by making the extension lower case right?

**Comment by daltonkell** [comment]
Yeah, I believe it should be an easy fix! There's also the possibility to add in some more robust checking, e.g. looking at the file header.

**Comment by Bobfrat** [comment]
so it does work if you rename the file?

**Comment by daltonkell** [comment]
It does work, yeah

**Comment by benjwadams** [comment]
You might want to check the first few bytes of the file (aka "magic" in Unix) parlance.

For example:
```
[badams@localhost]~/devel/glider-dac% file ~/Downloads/UW157-20141116T2118_f7fa_a808_6f48.nc
/home/badams/Downloads/UW157-20141116T2118_f7fa_a808_6f48.nc: NetCDF Data Format data
[badams@localhost]~/devel/glider-dac% head -c20 ~/Downloads/UW157-20141116T2118_f7fa_a808_6f48.nc
CDF
```

**Comment by daltonkell** [comment]
@benjwadams this is exactly what I had in mind!
